### DISCUSSION TITLE
ERD View: Settings > Data Model

### DISCUSSION THREAD

**Discussion by JonathanSchndr** [directus/directus#13359] [Feature Requests]
Moin! :wave:

I would find it handy to be able to view the current collections as ERD on the Data Model page. My suggestion would be to build a view with https://www.nomnoml.com/ (https://github.com/skanaar/nomnoml). What do you think about this?

If you think this is good, I would start with the development.  :rocket:

**Comment by benhaynes** [comment]
I love it, and can provide designs... but defer to Rijk on if this is a good one to have in Core.

**Comment by rijkvanzanten** [comment]
Love the idea, but before we dive into any development, lets discuss what it should look like, how it should work, and how it should be built 🙂 A couple things that immediately come to mind is: It should be interactive (eg allow reorganizing the view, clicking on the the collections to open the detail page, etc), should reuse our components to visually match the rest of the app, and should probably be an additional view to the data model page, rather than a replacement (as the current data model page also allows some organizational settings).

As for the recommended library, I don't think `nomnoml` is really the right tool for the job, as it is a SVG diagram renderer based on a custom input syntax. We would have to transform our own schema information into this string representation to then render it again into the diagram, which (from what I can tell) won't be interactive. It [would also add 75kb to the app](https://bundlephobia.com/package/nomnoml@1.5.1) for that overview, which is quite a lot. (I also generally prefer to reduce the number of third party libraries to a minimum, the difficulty of long term maintenance grows exponentially with the number of third party libs added)

@Nitwel already wrote some logic to draw arrows on proper paths for #12522, so I think we should be able to hand roll such a thing 🙂

**Reply by JonathanSchndr** [reply]
I had thought of nomnoml, since you can perhaps take it here as a transfer to export it. But I find #12522 also very good as a solution and it fits visually of course.

I think the 1.0 version could be first the pure representation and linking in the editing of the Collection. We are already in Settings > Data Model and could use this as another navigation / layout (instead of the table).

**Comment by Nitwel** [comment]
I like the idea too! We can probably add this feature with our v-workspace component, we just have to move the arrow logic to that component and add a bit for relation types.

**Comment by rokdd** [comment]
Just a update from the official Blog post of directus: [Community Spotlight: ERD Generator Extension for Visualizing Data Relationships](https://directus.io/blog/community-spotlight-erd-generator-extension-for-visualizing-data-relationships/)

As I understand everybody in the Cloud will get the extension.. Question is it also possible to use the extension for self hosted instances?

**Reply by rijkvanzanten** [reply]
Heya! This example highlights an extension somebody else made for themselves. We hope that it'll be public some day :) As for cloud extensions in general, we will be creating more premium extensions exclusive to paying customers as an incentive to support the project. We do offer a self-hosted container that includes those extensions (and any we'll create in the future) which can be acquired by contacting the sales team through https://directus.io/contact 🙂

**Reply by NilsBaumgartner1994** [reply]
We gave one of our student this ERD Generator as a thesis topic. It might be a bit more experimental, we will keep you informed.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
